lib/block/block.go

```diff
@@ -41,26 +42,74 @@ func (bck Block) String() string {
 	return string(encoded)
 }
 
-func MakeGenesisBlock(st *storage.LevelDBBackend, account BlockAccount) Block {
-	proposer := "" // null proposer
-	round := round.Round{
-		Number:      0,
-		BlockHeight: 0,
-		BlockHash:   base58.Encode(common.MustMakeObjectHash(account)),
-		TotalTxs:    0,
+// MakeGenesisBlock makes genesis block from genesis account and transaction.
+// The genesis block has different part from the other Block
+// * `Block.Proposer` is empty
+// * `Block.Round` is empty
+// * `Block.Confirmed` is `common.GenesisBlockConfirmedTime`
+// * has only one `Transaction`
+//
+// This Transaction is different from other normal Transaction;
+// * must have only one `Operation`, `OperationCreateAccount`
+// * `Transaction.B.Source` is same with `OperationCreateAccount.Target`
+// * `Transaction.B.Fee` is 0
+// * `OperationCreateAccount.Amount` is same with balance of genesis account
+// * `OperationCreateAccount.Target` is genesis account
+func MakeGenesisBlock(st *storage.LevelDBBackend, account BlockAccount, networdID []byte) (blk Block, err error) {
+	var exists bool
+	if exists, err = ExistsBlockByHeight(st, 1); exists || err != nil {
+		if exists {
+			err = errors.ErrorBlockAlreadyExists
+		}
+
+		return
+	}
+
+	// create create-account transaction.
+	opb := transaction.NewOperationBodyCreateAccount(account.Address, account.Balance)
+	op := transaction.Operation{
+		H: transaction.OperationHeader{
+			Type: transaction.OperationCreateAccount,
+		},
+		B: opb,
+	}
+
+	txBody := transaction.TransactionBody{
+		Source:     account.Address,
+		Fee:        0,
+		SequenceID: account.SequenceID,
+		Operations: []transaction.Operation{op},
+	}
+
+	tx := transaction.Transaction{
+		T: "transaction",
+		H: transaction.TransactionHeader{
+			Created: common.GenesisBlockConfirmedTime,
+			Hash:    txBody.MakeHashString(),
+		},
+		B: txBody,
 	}
-	transactions := []string{}
-	confirmed := ""
+	tx.Sign(kp, []byte(networdID))
+
+	transactions := []string{tx.GetHash()}
 
-	b := NewBlock(
-		proposer,
-		round,
+	blk = NewBlock(
+		"",
+		round.Round{}, // empty round
 		transactions,
-		confirmed,
+		common.GenesisBlockConfirmedTime,
```

Comment:
Why is the block confirmed time constant?

Reply:
@Charleslee522 Genesis block will be created in every node separately, so we need to set it to be same.